mos-platform/geos-cbm/geos_crt.c

```diff
@@ -0,0 +1,2013 @@
+}
+
+void RestartProcess(uint8_t process_num)
+{
+  asm volatile("jsr __RestartProcess" : : "x" (process_num) : "a","c","v");
+}
+
+void BlockProcess(uint8_t process_num)
+{
+  asm volatile("jsr __BlockProcess" : : "x" (process_num) : "a","c","v");
+}
+
+void UnblockProcess(uint8_t process_num)
+{
+  asm volatile("jsr __UnblockProcess" : : "x" (process_num) : "a","c","v");
+}
+
+void FreezeProcess(uint8_t process_num)
+{
+  asm volatile("jsr __FreezeProcess" : : "x" (process_num) : "a","c","v");
+}
+
+void UnfreezeProcess(uint8_t process_num)
+{
+  asm volatile("jsr __UnfreezeProcess" : : "x" (process_num) : "a","c","v");
+}
+
+void Sleep(uint16_t jiffies)
+{
+  __r0 = jiffies;
+  asm volatile("jsr __Sleep" : : : "a","x","y","c","v");
+}
+
+void EnableProcess(uint8_t process_num)
+{
+  asm volatile("jsr __EnableProcess" : : "x" (process_num) : "a","c","v");
+}
+
+/***********************************************************************
+ * GEOS Menu, Dialog Boxes & Icons routines
+ ***********************************************************************/
+
+uint8_t DoDlgBox(uint8_t *dlg_table)
+{
+  __r0 = (uint16_t)dlg_table;
+  asm volatile("jsr __DoDlgBox " : : : "a","x","y","c","v");
+  return __r0L;
+}
+
+void RstrFrmDialog(uint8_t icon_no)
+{
+  sysDBData = icon_no;
+  asm volatile("jsr __RstrFrmDialog " : : : "a","x","y","c","v");
+}
+
+void DoMenu(const menu_tab_t* menu_table, uint8_t selected_menu_no)
+{
+  __r0 = (uint16_t)menu_table;
+  asm volatile("jsr __DoMenu " : "=a" (selected_menu_no) : "a" (selected_menu_no) : "x","y","c","v");
+}
+
+void DoPreviousMenu(void)
+{
+  asm volatile("jsr __DoPreviousMenu " : : : "a","x","y","c","v");
+}
+
+void ReDoMenu(void)
+{
+  asm volatile("jsr __ReDoMenu " : : : "a","x","y","c","v");
+}
+
+void RecoverMenu(void)
+{
+  asm volatile("jsr __RecoverMenu " : : : "a","x","y","c","v");
+}
+
+void RecoverAllMenus(void)
+{
+  asm volatile("jsr __RecoverAllMenus" : : : "a","x","y","c","v");
+}
+
+void GotoFirstMenu(void)
+{
+  asm volatile("jsr __GotoFirstMenu" : : : "a","x","y","c","v");
+}
+
+void DoIcons(const icon_table_t* icon_table)
+{
+  __r0 = (uint16_t)icon_table;
+
+  asm volatile("jsr __DoIcons " : : : "a","x","y","c","v");
+}
+
+/***********************************************************************
+ * MISC & General Utitlities
+ ***********************************************************************/
+
+uint16_t GetRandom(void)
+{
+  asm volatile("jsr __GetRandom " : : : "a","c","v");
+  return random;
+}
+
+void CopyString(char *src, char *dest)
+{
+  __r0 = (uint16_t)src;
+  __r1 = (uint16_t)dest;
+  asm volatile(
+      "ldx   #2 \n"
+      "ldy   #4 \n"
+      "jsr   __CopyString " : : : "a","x","y","c","v");
+}
+
+void CopyFString(uint8_t *src, uint8_t *dest, uint8_t count)
+{
+  __r0 = (uint16_t)src;
+  __r1 = (uint16_t)dest;
+  asm volatile(
+      "ldx   #2 \n"
+      "ldy   #4 \n"
+      "jsr   __CopyFString " : "=a" (count) : "a" (count) : "x","y","c","v");
```

Comment:
No need to specify an output operand as `count` is passed by value and never leaves the function.

Reply:
Problem is that a is overwritten in __CopyFString. When i put a in input and clobber list i get an error message. Best way i found to tell compiler that a is overwritten was by putting it in the output list.

Any idea is welcome :-)

Reply:
Because I rember seeing generated code where the later code was dependend on the value in a. Thats why i would not like to remove it.

Reply:
I could not reproduce it so i remove the output operands where not needed.